### DISCUSSION TITLE
Problem changing image from monochrome to colour

### DISCUSSION THREAD

**Discussion by dormant** [ImageMagick/ImageMagick6#313] [General] [answered]
### ImageMagick version

6.9.12-98

### Operating system, version and so on

Debian GNU/Linux trixie/sid

### Description

I am batch processing a bunch of image files using mogrify. The images contain black lines, which I want to thicken. 
![0-orig](https://github.com/ImageMagick/ImageMagick/assets/156362/1c88b82c-95d9-40cb-b7ba-366dbb853078)

As the pixels are actually grayscale pixels, my processing goes like this:

```
mogrify -grayscale Rec709Luma *.png
mogrify -black-threshold 75% *.png
mogrify -convolve '3x3:1,1,1,1,1,1,1,1,1' *.png
mogrify -monochrome *.png
```

And this is the result:
![1-thick](https://github.com/ImageMagick/ImageMagick/assets/156362/2ac0529e-876b-4361-b78a-aeccd2a8a899)

For some of the files, I want to convert black to another colour. I tried this:

```
mogrify -colorspace RGB *.png
mogrify -opaque black -fill red  *.png
```

But I get a pure white image:
![2-red](https://github.com/ImageMagick/ImageMagick/assets/156362/76638201-c932-47bb-955a-f11918555946)

The image seems to be grayscale, not RGB.

How can I do this last step?

**Comment by fmw42** [comment, marked as answer]
In place of -colorspace RGB, use PNG24:*.png to force 24bit color RGB.  See if that works.

Also if you want to thicken your line, you could simply use -morphology open diamond:X Then try varying X=1,2,3...etc until you get the thickness you want.  You can also try disk:X which will be thicker for any give X, which can be float.

**Reply by dormant** [reply]
Many thanks, this is very helpful. The -morphology option is exactly what I need.
